codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -341,4 +395,12 @@ public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
         super.handlerRemoved(ctx);
         cleanup();
     }
+
+    public final int blockSize() {
+        return blockSize;
+    }
+
+    public final int currentBlockLength() {
+        return currentBlockLength;
```

Comment:
I've simply, perhaps naively, used an existing field to check the state of the class. I'll look at the use of that field, as well, but hadn't thought about it too much, tbh.